### PR TITLE
Underscore is a valid name character

### PR DESCRIPTION
**Pull request by alexiri**
Underscores are allowed according to: https://github.com/esphome/esphome/blob/dev/esphome/const.py#L9

### CODE REVIEW

File: src/wizard/wizard-dialog.ts [hunk 1]
```diff
@@ -181,8 +181,8 @@ export class ESPHomeWizardDialog extends LitElement {
         label="Name"
         name="name"
         required
-        pattern="^[a-z0-9-]+$"
-        helper="Only use lowercase letters (a-z), numbers (0-9) or dash (-)"
+        pattern="^[a-z0-9-_]+$"
+        helper="Only use lowercase letters (a-z), numbers (0-9), dash (-) or underscore (_)"
         @input=${this._cleanNameInput}
```
Comment (balloob) [inline src/wizard/wizard-dialog.ts:186]: This method will still replace `_` with `-`.

Comment (alexiri) [reply src/wizard/wizard-dialog.ts:186]: Sorry, missed that!

**Comment by frenck** [comment]
The reason the underscore is still accepted is for legacy. As the name is used as hostname as well, the underscore is actually not valid.

For example: `http://my_test.local` isn't a valid domain. In many cases, it will gracefully work, but isn't according to specifications and can (and has) actually caused problems.

Hence, it isn't accepted for newer devices at this point.

IMHO this change should not be merged for that reason.

**Comment by alexiri** [comment]
Fair enough. As it was previously valid I thought this was a regression, and not a conscious choice.

**Pull request closed without merging**